internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,52 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Check if the resource version has changed
+	if pdbWatcher.Status.ResourceVersion != "" && pdbWatcher.Status.ResourceVersion != deployment.ResourceVersion {
+		// Update ResourceVersion and MinReplicas
+		pdbWatcher.Status.ResourceVersion = deployment.ResourceVersion
+		pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
+		err = r.Status().Update(ctx, pdbWatcher)
+		if err != nil {
+			logger.Error(err, "Failed to update PDBWatcher status")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Track initial state to detect conflicts
+	initialResourceVersion := deployment.ResourceVersion
+	initialReplicas := *deployment.Spec.Replicas
+
 	// Update status dynamically
-	pdbWatcher.Status.CurrentReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MaxReplicas = *deployment.Spec.Replicas + deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
-	pdbWatcher.Status.ScaleFactor = 1 // Dynamically Calculated
+	if pdbWatcher.Status.CurrentReplicas == 0 {
+		pdbWatcher.Status.CurrentReplicas = initialReplicas
+	}
+	pdbWatcher.Status.MinReplicas = initialReplicas
+
+	// Handle nil Deployment Strategy and MaxSurge
+	maxSurge := int32(1) // Default max surge value
+	if deployment.Spec.Strategy.RollingUpdate != nil && deployment.Spec.Strategy.RollingUpdate.MaxSurge != nil {
+		if deployment.Spec.Strategy.RollingUpdate.MaxSurge.Type == intstr.Int {
+			maxSurge = deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
+		} else if deployment.Spec.Strategy.RollingUpdate.MaxSurge.Type == intstr.String {
+			percentageStr := strings.TrimSuffix(deployment.Spec.Strategy.RollingUpdate.MaxSurge.StrVal, "%")
+			percentage, err := strconv.Atoi(percentageStr)
+			if err == nil {
+				maxSurge = (initialReplicas * int32(percentage)) / 100
+			}
+		}
+	}
+	pdbWatcher.Status.MaxReplicas = initialReplicas + maxSurge
+
+	pdbWatcher.Status.ScaleFactor = 1 // Modify based on requirements
+
+	// Save ResourceVersion to PDBWatcher status
+	pdbWatcher.Status.ResourceVersion = initialResourceVersion
```

Comment:
this is the deployments resource version so you don't want to change it until your controller has mutated the deployment  If this is the create its hadnled way up at 131.

Reply:
Rather you want to save teh resource version after this line

deployment.Spec.Replicas = &newReplicas
			err = r.Update(ctx, deployment)